runtime/amd64.S

```diff
@@ -65,6 +65,14 @@
 
 #endif
 
+#if defined(SYS_linux) || defined(SYS_gnu)
+#define ENDFUNCTION(name) \
+        .type name,@function; \
```

Comment:
Isn't the `.type` directive already part of the `FUNCTION` macro ?

Reply:
Ah, yes, it is - for amd64 (but not for i386). I have just updated the patch. Thanks!